java/src/Ice/src/main/java/com/zeroc/Ice/Current.java

```diff
@@ -216,34 +216,32 @@ private OutgoingResponse createOutgoingResponseCore(Throwable exc) {
         throw new MarshalException("Unexpected exception type");
       }
 
-      // TODO: RequestFailedException's id, facet and operation should be read-only.
-      if (rfe.id.name.isEmpty()) {
-        rfe.id = id;
-        rfe.facet = facet;
+      Identity id = rfe.id;
```

Comment:
I find defining local variables that hide fields pretty confusing and unexpected, especially in a big method like this one.

Reply:
I'm open to changing it, but this is what we're doing in every language, so I'd like to change it in all of them.
See #2345

Reply:
In other languages, this code is not in Current, but in a separate class. So id/facet/operation don't hide fields.

Reply:
Fair enough, I added a prefix in Java to disambiguate them.